src/api.c

```diff
@@ -537,3 +551,64 @@ API int seccomp_export_bpf(const scmp_filter_ctx ctx, int fd)
 
 	return 0;
 }
+
+/* NOTE - function header comment in include/seccomp.h */
+int seccomp_alloc_notifications(struct seccomp_notif **req,
+				struct seccomp_notif_resp **resp)
+{
+	static struct seccomp_notif_sizes sizes = {};
+
+	if (sizes.seccomp_notif == 0) {
+		int ret = syscall(__NR_seccomp, SECCOMP_GET_NOTIF_SIZES, 0, &sizes);
+		if (ret < 0)
+			return errno;
+	} else {
+		return -1;
+	}
+
+	*req = malloc(sizes.seccomp_notif);
+	if (!*req)
+		return -ENOMEM;
+	memset(*req, 0, sizes.seccomp_notif);
+
+	*resp = malloc(sizes.seccomp_notif_resp);
+	if (!*resp) {
+		free(req);
+		return -ENOMEM;
+	}
+	memset(*resp, 0, sizes.seccomp_notif_resp);
+
+	return 0;
+}
+
+/* NOTE - function header comment in include/seccomp.h */
+void seccomp_free_notifications(struct seccomp_notif *req,
+				struct seccomp_notif_resp *resp)
+{
+	free(req);
+	free(resp);
+}
+
+/* NOTE - function header comment in include/seccomp.h */
+int seccomp_receive_notif(int fd, struct seccomp_notif *req)
+{
+	if (ioctl(fd, SECCOMP_IOCTL_NOTIF_RECV, req) < 0)
```

Comment:
This may need to move to "src/system.c" in order to work better with the "API levels" functionality.  For example, what happens if a user calls `seccomp_recerive_notif()` with an API level of "1"?